mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraSchemaVersionManagerTest.java

```diff
@@ -0,0 +1,204 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.cassandra.mail;
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.mockito.Mockito;
+import uk.org.lidalia.slf4jtest.TestLogger;
+import uk.org.lidalia.slf4jtest.TestLoggerFactory;
+
+import java.util.concurrent.CompletableFuture;
+
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.TOO_OLD;
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.TOO_RECENT;
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.OK;
+import static org.apache.james.mailbox.cassandra.mail.CassandraSchemaVersionManager.SchemaVersionState.OLD_BUT_SUPPORTED;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.org.lidalia.slf4jtest.LoggingEvent.info;
+import static uk.org.lidalia.slf4jtest.LoggingEvent.warn;
+
+public class CassandraSchemaVersionManagerTest {
+
+    private final int minVersion = 2;
+    private final int maxVersion = 4;
+    private CassandraSchemaVersionDAO schemaVersionDAO;
+    TestLogger logger = TestLoggerFactory.getTestLogger(CassandraSchemaVersionManager.class);
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void setUp() {
+        schemaVersionDAO = Mockito.mock(CassandraSchemaVersionDAO.class);
+    }
+
+    @After
+    public void tearDown() {
+        TestLoggerFactory.clear();
+    }
+
+    @Test
+    public void getVersionTestShoudReturnToOldIfVersionIsToOld() {
+        int currentVersion = minVersion - 1;
```

Comment:
I don't. Like I did make it obvious to the reader I want something just before minVersion